### ISSUE TITLE
Hardcoded Tray Icons

### ISSUE THREAD

**Issue by Foggalong** [hexchat/hexchat#1462]
The tray icons in hexchat are hardcoded which means they can't be themed by icon themes. The way they are hardcoded though means that tools such as [this one](https://github.com/bil-elmoussaoui/Hardcode-Tray) can't undo the hardcoding and allow themes to do their thing.

I know there's an appindicator branch which was possibly working to fix issues related to this, but it's been active for quite a few months now. Wondered if there was any word on this?

**Comment by TingPing** [comment]
They are not hardcoded - https://hexchat.readthedocs.org/en/latest/appearance.html#icons

The branch you mentioned just respected icons in the normal paths it has always been theme-able.

**Comment by Foggalong** [comment]
That just means that the second sentence of my first paragraph wasn't right - the icons are still hardcoded. The default icons should be located in `/usr/share/icons` and then the standard GTK loopup implemented such that if the relevant icons exist in the users current theme they're picked up from there.

**Comment by TingPing** [comment]
I define hardcoded as not changeable which they are, hexchat just doesn't look up icons in any of the standard folders.

**Comment by Foggalong** [comment]
They are changeable, but not by the icon themes which is the important bit.

**Comment by TingPing** [comment]
Ok, As mentioned https://github.com/hexchat/hexchat/commit/320e0582406579e975806d4289b271cc521391cf is the start of that support.

**Comment by Foggalong** [comment]
Yeah I noticed that and the last commit was over 6 months ago which was why I was checking in :)

**Comment by geerds** [comment]
Would be great to have this in hexchat!

**Comment by Foggalong** [comment]
I've actually just tried the method from the docs for getting custom icons but it doesn't seem to be working. This is the directory structure I have 

```
~/.config/hexchat/icons
├── hexchat.svg
├── tray_fileoffer.svg
├── tray_highlight.svg
└── tray_message.svg
```

**Comment by geerds** [comment]
I've encountered the same problem

**Comment by Foggalong** [comment]
For when the standard icon paths get implemented, there's also a problem with using the name `hexchat.svg` as the icon name. This makes it impossible for icon theme developers to include the status icon without accidentally overwriting the app icon elsewhere. This is okay while it's using it's own themeing config but when this is implemented I'd recommend renaming the icon to `hexchat-tray.svg`

**Comment by TingPing** [comment]
@Foggalong If you looked at the commit linked above it does this.

**Comment by TingPing** [comment]
>I've actually just tried the method from the docs for getting custom icons but it doesn't seem to be working.

@Foggalong Because you misread the docs, it only supports png.

**Comment by Foggalong** [comment]
@TingPing Sorry, I'd looked at [this bit](https://github.com/hexchat/hexchat/blob/320e0582406579e975806d4289b271cc521391cf/data/icons/Makefile.am#L1) and not seen the icon mentioned but didn't see [this bit where it is](https://github.com/hexchat/hexchat/blob/320e0582406579e975806d4289b271cc521391cf/src/fe-gtk/plugin-tray.c#L49-L52).

I'd incorrectly assumed that the `.png` was just an example being as it's the most popular icon image extension. It should probably say explicitly in the docs that png is a requirement rather than a recommendation like the 16x16 size. [Is this the right file](https://github.com/hexchat/documentation/blob/master/appearance.rst) to submit a pull request for?

**Comment by TingPing** [comment]
>  Sorry, I'd looked at this bit and not seen the icon mentioned but didn't see this bit where it is.

Icon themes work in layers, so it looks up `hexchat-tray-something` and if not found looks up `hexchat-tray` and if not found looks up `hexchat`. So installing hexchat.svg but allowing themes to add hexchat-tray.svg works.

>  Is this the right file to submit a pull request for?

Yes.

**Comment by Foggalong** [comment]
That's a good system to have, yeah. And thanks, I'll send that pull request along asap

**Comment by Foggalong** [comment]
Pull opened hexchat/documentation#25
